### PR TITLE
Change made Tick-Tack-Toe future proof

### PR DESCRIPTION
**Pull request by gizzmo123456**

### CODE REVIEW

File: oxo.py [hunk 1]
```diff
@@ -1,26 +1,195 @@
 class OxoBoard:
-    def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+
+    PLAYER_y = 0
+    PLAYER_X = 1
```
Comment (edpowley) [inline oxo.py:4]: Are these used?

File: oxo.py [hunk 2]
```diff
@@ -1,26 +1,195 @@
...
+
+    def __init__(self, grid_width=3, grid_height=3, win_row_amount=3):
+        """ The initialiser. Initialise any fields you need here.
+
+        :param grid_width:     width of board
+        :param grid_height:    height of board
+        :param win_row_amount:
+        """
+        self.grid_size = {"width": grid_width, "height": grid_height}
```
Comment (edpowley) [inline oxo.py:13]: A dictionary probably isn't the best choice of data structure here -- I'd recommend using a 2-tuple `(grid_width, grid_height)` or two fields `self.grid_width` and `self.grid_height`

File: oxo.py [hunk 3]
```diff
@@ -1,26 +1,195 @@
...
+            winner = self.check_vert(player_index)
+            if winner != 0:
+                return winner
+            # Diagonals
+            winner = self.check_diagonals(player_index)
+            if winner != 0:
+                return winner
+        return 0
+
+    def check_hoz(self, player):
```
Comment (edpowley) [inline oxo.py:57]: `check_hoz` and `check_vert` are quite similar -- could they be combined?

File: oxo.py [hunk 4]
```diff
@@ -1,26 +1,195 @@
...
+            for x in range(self.grid_size["width"]):
+
+                # break if its not possible to get the required amount it a row
+                if count_in_row == 0 and self.grid_size["width"] - x < self.win_row_amount:
+                    break
+
+                # reset and continue if we have no input
+                if (x, y) not in self.grid:
+                    count_in_row = 0
+                    continue
```
Comment (edpowley) [inline oxo.py:79]: This all looks ok, however be wary of using `break` and `continue` too much as they can make code hard to follow. This could have been rewritten just with nested `if` statements for example.

**Review by edpowley — COMMENTED** [review]
Good - all tests passed, good commenting, good expandability. However your code is a little long and repetitive -- would be worth finding ways to refactor and reuse.

File: oxo.py (hunk 3 above)
Comment (gizzmo123456) [reply oxo.py:57]: On a reflection of this I could of added 'X_size' and 'Y_size' to the function pramas and flip the values when called for instance 
check_row(self.grid_size["width"], self.grid_size["height"], player) for horizontal and check_row(self.grid_size["height"], self.grid_size["width"], player) for vertical

I would also have to change some of the variables in the function to the pramas as well but it would be pretty straight forward
